src/App.js

```diff
@@ -1,25 +1,255 @@
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    // holds income categories and values
+    const [income, setIncome] = useState(initalIncome);
+    // holds expense categories
+    const [expenses, setExpenses] = useState(initialExpenses);
+    // holds the values of the expenses in each category
+    const [expenseValues, setExpenseValues] = useState(initialExpenseValues);
+
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [userId, setUserId] = useState("");
+
+
+    // on component mount...
+    useEffect(() => {
+        // if the user has previously logged in and their userId is stored in localStorage...
+        if (window.localStorage.userId) {
+            // set the userId in state to the userId of the previous session
+            // and change isLoggedIn to true
+            setUserId(window.localStorage.userId);
+            setIsLoggedIn(true);
+        }
+    }, []);
+
+    // when userId is updated...
+    useEffect(() => {
+        // if window.localStorage exists...
+        if (window.localStorage) {
+            // store the userId in localStorage
+            window.localStorage.userId = userId;
+        }
+
+        // if no userId exists in state...
+        if (!userId) {
+            // do nothing else
+            return;
+        }
+
+        // reference the user's data stored in the database
+        const database = getDatabase(firebase);
+        const dbRef = ref(database, `${userId}/`);
+
+        // when the user logs in & when the user's data is updated...
+        onValue(dbRef, (response) => {
+            // if the user's database exists and has values
+            if (response.exists()) {
+                // use the data returned from the database to set the income and expense state variables
+                const data = response.val();
+                setIncome(data.income);
+                setExpenses(data.expenses);
+                setExpenseValues(data.expenseValues);
+            }
+        });
+    }, [userId]);
+
+
+    const handleInputChange = (e, needKey = true) => {
+        // initialize value variable
+        let value;
+
+        // if the user has not deleted all numerical values from the target input
+        if (e.target.value !== "$") {
+            // set the value variable to the value of the target input after removing commas and dollar signs
+            value = Number(e.target.value.replace(/[,$]/g, ''));
+        } 
+        
+        // if the target input has no numerical value
+        if (!value) {
+            // set the value target to 0
+            value = 0;
+        }
+        
+        // if the numerical value of the target input exceeds the value cap
+        if (value > 99999) {
```

Comment:
I'm assuming that the app is used for monthly expense tracking, so this restriction is reasonable. But if someone wanted to do a yearly tracking, then 6-digit salary/wages is not possible. Maybe you could show on the webpage that it's a monthly-type tracker? Or have an option somewhere that lets the user switch between the two monthly/annual modes so that this limit can change

Reply:
that's a really good catch, thank you!! i'm hoping to eventually use the sidebar to keep track of months so that each user can have multiple months, but I hadn't thought about making it more clear in this version that this is a monthly tracker - this also gives me some good ideas for extra stretch goals to work on after bootcamp, so thank you for that too!! 💫